### PR TITLE
Added unit test cases for Gauge chart

### PR DESCRIPTION
**Pull request by v-baambati**
1. Added unit test cases
2. Added unit test plan

### CODE REVIEW

File: packages/react-charting/UnitTests/GaugeChartUT.test.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  test('Should return zero arcs when there is no data', () => {
+    const instance = new GaugeChartBase(data);
+    expect(instance).toBeDefined();
+    const result = instance._getMargins();
+    expect(result.bottom).toEqual(16);
+    expect(result.top).toEqual(18);
+    expect(result.left).toEqual(56);
+    expect(result.right).toEqual(56);
+  });
+});
```
Comment (srmukher) [inline packages/react-charting/UnitTests/GaugeChartUT.test.tsx:50]: Add unit tests for calcNeedleRotation, getSegmentLabel, getChartValueLabel, _getStylesBasedOnBreakpoint, _wrapContent, _yValueHoverSubCountsExists

Comment (v-baambati) [reply packages/react-charting/UnitTests/GaugeChartUT.test.tsx:50]: Sure @srmukher

File: packages/react-charting/UnitTests/GaugeChartUT.test.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  test('Should return zero arcs when there is no data', () => {
+    const instance = new GaugeChartBase(emptyData);
+    expect(instance).toBeDefined();
+    const result = instance._processProps();
+    expect(result.arcs).toHaveLength(0);
+  });
+});
+
+runTest('_getMargins', () => {
```
Comment (srmukher) [inline packages/react-charting/UnitTests/GaugeChartUT.test.tsx:40]: Write all possible cases considering combinations of hideMinMax, chartTitle, sublabel are true/false

File: packages/react-charting/UnitTests/GaugeChartUT.test.tsx (hunk 1 above)
Comment (v-baambati) [reply packages/react-charting/UnitTests/GaugeChartUT.test.tsx:405]: Added test cases for mentioned functions

**Review by Hotell — APPROVED** [review]

**Comment by AtishayMsft** [comment]
@v-baambati Could you pull latest master for this PR

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/1db129b125726aca0f500ac1f292eca59e7db57b'>1db129b125726aca0f500ac1f292eca59e7db57b</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-charting/UnitTests/GaugeChartUT.test.tsx [hunk 3]
```diff
@@ -0,0 +1,405 @@
+import { GaugeChartVariant, GaugeValueFormat, IGaugeChartProps } from '../src/components/GaugeChart/index';
```
[the remark this answers is not in the document]
Comment (AtishayMsft) [reply packages/react-charting/UnitTests/GaugeChartUT.test.tsx:1]: The candidate screenshot is the correct representation of vertical bar chart date axis. Fixed in a PR #30691

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v8"/>

**Review by AtishayMsft — APPROVED** [review]

**Pull request merged**